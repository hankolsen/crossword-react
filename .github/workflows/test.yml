name: melodikryzzet-tests

on: 
  push:
    branches:
      - 'master'
  pull_request:


jobs:
  test:
    name: 'Node ${{ matrix.node }}'
    runs-on: ubuntu-latest
    strategy:
      matrix:
        node: [12, 14]
    env:
      SKIP_PREFLIGHT_CHECK: true
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v1
      with: 
        node-version: ${{ matrix.node }}
    - name: Install dependencies
      uses: bahmutov/npm-install@v1
      with: 
        install-command: yarn
    - name: Run tests
      run: yarn test:ci