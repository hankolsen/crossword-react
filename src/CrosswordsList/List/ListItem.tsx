import React from 'react';
import { Link } from 'react-router-dom';

type Props = {
  crossword: {
    id: string;
    name: string;
  };
};

const ListItem = (props: Props) => {
  const {
    crossword: { id, name },
  } = props;

  return (
    <div className="list__list-item">
      <Link className="list-item__button" to={`/crossword/${id}`} key={id}>
        {name}
      </Link>
    </div>
  );
};

export default ListItem;
