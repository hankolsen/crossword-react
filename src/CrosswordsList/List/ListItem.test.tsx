import React from 'react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { screen, render } from 'test/test-utils';
import ListItem from './ListItem';

describe('ListItem test', () => {
  it('should render a list item', () => {
    const crossword = {
      id: 'abc-123',
      name: 'The Crossowrd to rule them all',
    };
    const { container } = render(
      <BrowserRouter>
        <ListItem crossword={crossword} />
      </BrowserRouter>,
    );
    expect(
      screen.queryByText(/the crossowrd to rule them all/i),
    ).toBeInTheDocument();
    expect(container.querySelector('a')?.getAttribute('href')).toBe(
      '/crossword/abc-123',
    );
    expect(container).toMatchSnapshot();
  });
});
