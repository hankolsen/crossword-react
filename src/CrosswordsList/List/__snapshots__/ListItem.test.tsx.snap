// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem test should render a list item 1`] = `
<div>
  <div>
    <div
      class="list__list-item"
    >
      <a
        class="list-item__button"
        href="/crossword/abc-123"
      >
        The Crossowrd to rule them all
      </a>
    </div>
  </div>
</div>
`;
